### PR TITLE
Fix peacock escalations' incomplete displayed killing tips

### PR DESCRIPTION
**Pull request by suanjiansalt**
Improvement program for issue #353 , huge thanks to @grappigegovert .

A dumb try, so please treat this as a pupil's homework, after all it's my first time doing this...

Note: 

- [x] Here I also did some adjustments to `The Christmas Calamity` gameplay. <s>If it's broken, I'm appreciated to receive your kind advice.</s> This time I fixed it correctly. If it's unnecessary, I will rollback it in no time.
- [x] Icon shown by `HUDTemplate` line looks improper in `The Tedious Thievery`. Before you collect all items (7 golds and 2 burgers) the icon is exactly the 'Exit' icon, but after completion the icon is what we want (a blue 'objective' icon). <s>I deleted `OnActive` line but this could be ridiculous. Please inform me before I look too stupid.</s> I changed the `iconType` to `16` and it looks good now.
- [x] Maybe all targets' name localization is required to realize?

### CODE REVIEW

File: contractdata/PARIS/ESCALATION/XMAS/XMAS1.json [hunk 1]
```diff
@@ -76,7 +80,12 @@
                                             },
                                             {
                                                 "$inarray": {
-                                                    "in": "$.LethalWeapons",
+                                                    "in": {
+                                                        "$or": [
```
Comment (grappigegovert) [inline contractdata/PARIS/ESCALATION/XMAS/XMAS1.json:84]: Does this work?

**Review by AnthonyFuller — APPROVED** [review]
RSLGTM for l10n. A note: some lines that were previously translated to Chinese are no longer translated.

File: contractdata/PARIS/ESCALATION/XMAS/XMAS1.json (hunk 1 above)
Comment (suanjiansalt) [reply contractdata/PARIS/ESCALATION/XMAS/XMAS1.json:84]: Sorry, but maybe I used too much imagination and creativity here. You may see that all I want to do is to take Christmas explosives into consideration, so maybe there's a more acceptable type exist?

Comment (grappigegovert) [reply contractdata/PARIS/ESCALATION/XMAS/XMAS1.json:84]: You could just add the two repo ids of the explosives to the `LethalWeapons` array. I'm not entirely sure if explosive weapons can be checked like this however, so I suggest testing it.

Comment (suanjiansalt) [reply contractdata/PARIS/ESCALATION/XMAS/XMAS1.json:84]: That makes sense... So the final question is, how to test? 🤯

Comment (grappigegovert) [reply contractdata/PARIS/ESCALATION/XMAS/XMAS1.json:84]: Play the contract and check if killing a target with your added explosives changes the counter?

Comment (suanjiansalt) [reply contractdata/PARIS/ESCALATION/XMAS/XMAS1.json:84]: I mean how to replace the game file with my JSON

**Comment by suanjiansalt** [comment]
Decide to localize all killing tip strings. Sorry for the wavering.

File: contractdata/PARIS/ESCALATION/XMAS/XMAS1.json (hunk 1 above)
Comment (grappigegovert) [reply contractdata/PARIS/ESCALATION/XMAS/XMAS1.json:84]: You can either follow the steps [in the Readme.md under "Installation"](https://github.com/thepeacockproject/Peacock?tab=readme-ov-file#installation), and just run a dev version of Peacock with your modified files, or if you want you could copy the contract json into the "contracts" folder of your release version of Peacock, and then access the contract through "My contracts" in game.

**Review by AnthonyFuller — APPROVED** [review]
I assume it all works and has been verified.

**Comment by grappigegovert** [comment]
All of these "Eliminate <name>" lines without any condition don't really need dedicated localization and create unnecessary work for the translators, the generic UI_CONTRACT_GENERAL_OBJ_KILL should work fine

I've only really looked at the the first contract in the list, MENDIE1, and that one already was wrong: your new edit says that the kill needs to be an explosion accident, where any kill method is sufficient.
It also looks like you changed the locr key `UI_PEACOCK_SOUTHERN_COMFORT_OBJ1` to `UI_PEACOCK_SOUTHERN_COMFORT_XMASGIFT`, but in doing so changed the text for all languages to the english translation.

I'll do some more testing later

**Comment by suanjiansalt** [comment]
> All of these "Eliminate <name>" lines without any condition don't really need dedicated localization and create unnecessary work for the translators, the generic UI_CONTRACT_GENERAL_OBJ_KILL should work fine
1. I did this because in H3, almost official escalations really localized all strings like "Eliminate <name>". For example, in _the Jinzhen Incident_:
Screenshot 2024-01-02 110035
Just eliminating Shihong Luo and Hui Hou in any conditions are both localized. Again, as I said earlier, if it's not necessary, just delete it.
> I've only really looked at the the first contract in the list, MENDIE1, and that one already was wrong: your new edit says that the kill needs to be an explosion accident, where any kill method is sufficient.
2. Oh YES, it is my fault. Not understand why I did it, but it clearly a fault I made on purpose.
> It also looks like you changed the locr key `UI_PEACOCK_SOUTHERN_COMFORT_OBJ1` to `UI_PEACOCK_SOUTHERN_COMFORT_XMASGIFT`, but in doing so changed the text for all languages to the english translation.
3. Well, I'm not merely replace the whole line with new key `UI_PEACOCK_SOUTHERN_COMFORT_XMASGIFT`. All I did in fact is change the keys' names, so I also give the right value to the keys if you look into it. This happens to some complications of _The Jeffery Confession_ too.

**Comment by grappigegovert** [comment]
Alright, did a quick test with an xmax explosive, seems to work nicely.

The tileimage for the christmas calamity is broken with this change however, it would need to be updated in the image pack.

**Review by grappigegovert — APPROVED** [review]
Looks all good to me :)

**Comment by RDIL** [comment]
Thanks @suanjiansalt!

**Pull request merged**